internal_ws/ykpack/src/build.rs

```diff
@@ -0,0 +1,150 @@
+//! Various utilities for building SIR.
```

Comment:
So all of this stuff was lifted from the llvm codegen, but you'd like to be able to re-use it in the cranelift backend? I think that's right.

If so, perhaps a brief explanation of this in the doc comment?

Something like:
```
This code is intended for consumption by Rust code generator backends. By making these utilities public avoid the need for code duplication in each backend that wishes to generate SIR.
```